packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -283,6 +303,8 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
     const action = permission.action ?? 'lambda:InvokeFunction';
     const scope = permission.scope ?? this;
 
+    this.addWarningOnInvokeFunctionPermissions(scope, action);
```

Comment:
This looks like it will add the warning *always*, even if you're not using aliases or versions at all. Do I have that right?

If so, that is probably overly spammy. I'd prefer adding the warning if we detect the following:

- `grantInvoke` is called on the function; AND 
- `currentVersion` is called/has been called

If combine those, you're probably doing something wrong.

Reply:
I am happy to do this, but what about the scenarios that this will miss? I'm thinking:

```ts
const fn = new lambda.Function();
const version = new lambda.Version(this, 'v', {
  lambda: fn,
});
fn.grantInvoke();
```

Shoudn't we warn users who do this that they should grantInvoke on the version?

Reply:
To minimize the amount of support requests, I'd prefer to err on the side of caution: tell people when they're definitely doing something wrong, rather than when they *might* be doing something wrong.

So: I don't want false positives. We can talk about how to reduce false negatives.

Reply:
There might be ways, we might care to invest in them (we might also not).

Reply:
Okay! I think I've done exactly as you've described. Moved this from reducing false negatives to reducing false positives.

There is an `addVersion()` on Functions that is `@deprecated` in favor of `currentVersion` so I decided not to warn those users. I think `@deprecated` is signal enough that you should not be using it. But open to feedback here.